docs/aurora/workflows/parsl.md

```diff
@@ -1 +1,341 @@
 # Parsl on Aurora
+
+[Parsl](https://parsl.readthedocs.io/en/stable/) is a parallel programming library for Python.  It can be used to deploy large numbers of tasks in parallel and with complex dependencies on ALCF machines, and is particularly well suited to run high-throughput workflows.  While Parsl is a Python library, it can execute tasks that run any compiled application.  Parsl can also execute tasks that run mpi applications.
+
+Parsl uses Python's concurrent futures module to create functions that return a Python futures object.  A Parsl workflow operates by creating futures for tasks that the Parsl executor will then fulfill by running them on available compute resources.
+
+A Parsl workflow contains two parts:
+* the workflow logic of applications, tasks and task dependencies
+* the configuration of compute resources that execute tasks
+
+Here we sketch out some possible configurations for executing workflows on Aurora.  These docs were written for Parsl 2025.1.13.
+
+## Installation and Setup
+Parsl is a Python library and can be installed with `pip`.  For example, in a Python virtual environment:
+
+```bash linenums="1"
+python -m venv $HOME/_env
+source $HOME/_env/bin/activate
+pip install parsl
+```
+
+When using Parsl to distribute work over many PBS Jobs (first two examples below), your workflow script will be executed on a login node and will not return until all tasks are completed.  In this situation, it is advisable to run your script in a [screen](https://linuxize.com/post/how-to-use-linux-screen/) session on the login node.  To setup your screen session and run your workflow you can follow these steps:
+
+```bash linenums="1"
+screen -S parsl_session_name
+source _env/bin/activate
+python my_parsl_workflow.py
+# the program begins execution, waiting for compute resources and for tasks to complete. While it runs, you can disconnect from the screen session with:
+# Ctrl+a d
+
+# Some time later, reconnect to your screen session on the same login node with 
+screen -r parsl_session_name
+# progress of your workflow script will then be displayed again
+```
```

Comment:
```suggestion
```bash linenums="1"
screen -S parsl_session_name
source _env/bin/activate
python my_parsl_workflow.py
```
The program then begins execution, waiting for compute resources and for tasks to complete, and blocking the interactive shell. While it runs, you can disconnect from the screen session with ++ctrl+a++ ++d++ (assuming that GNU Screen is configured to use the command character ++a++).

Some time later, reconnect to your screen session on the same login node with:
```bash linenums="1"
screen -r parsl_session_name
```
Progress of your workflow script will then be displayed again.
```

this is mostly an excuse for me to use the Keys markup

Reply:
I think I need to manually apply this